src/vrapp/VRVolumeApp.cpp

```diff
@@ -1098,13 +1124,60 @@ void VRVolumeApp::update_dynamic_slices()
   }
 }
 
-void VRVolumeApp::do_grab(glm::mat4& newPose)
+void VRVolumeApp::do_grab(glm::mat4 &newPose)
 {
-  if (m_grab) {
+  if (m_grab)
+  {
     // Update the paintingToRoom transform based upon the new transform
     // of the left hand relative to the last frame.
     m_object_pose = newPose * glm::inverse(m_controller_pose) * m_object_pose;
   }
   m_controller_pose = newPose;
 }
 
+ArcBallCamera &VRVolumeApp::get_trackball_camera()
+{
+  return m_trackball;
+}
+
+void VRVolumeApp::set_animation_speed(float time)
+{
+  m_animation_speed = time;
+}
+
+Simulation &VRVolumeApp::get_simulation()
+{
+  return *m_simulation;
+}
+
+void VRVolumeApp::set_clip_min(glm::vec3 clip_min)
+{
+  m_ui_view->set_clip_min(clip_min);
+}
+
+void VRVolumeApp::set_clip_max(glm::vec3 clip_max)
+{
+  m_ui_view->set_clip_max(clip_max);
+}
+
+std::string VRVolumeApp::get_movie_state_label()
+{
+  if (m_current_movie_state == STOP)
+  {
+    return "Write Movie";
+  }
+  else
+  {
+    return "STOP RECORD";
+  }
+}
+
+MOVIESTATE VRVolumeApp::get_movie_state()
+{
+  return m_current_movie_state;
+}
+
+void VRVolumeApp::set_app_mode(APPMODE mode)
+{
+  m_app_mode = mode;
+}
```

Comment:
`VRVolumeApp.cpp` is also getting super long, worth thinking through how it could be refactored/abstracted into multiple files